Deprecate legacy doc (#18585)
nrfconnect/sdk-nrf: doc/nrf/libraries/security/nrf_security/doc/configuration.rst

PSA crypto support is provided through PSA Crypto APIs and is implemented by PSA core.
PSA core uses PSA drivers to implement the cryptographic features either in software, or using hardware accelerators.

.. _legacy_crypto_support:

Legacy crypto support
*********************

To enable the legacy crypto support mode of nRF Security, set the :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig option along with additional configuration options, as described in :ref:`nrf_security_legacy_config`.
The legacy crypto support allows backwards compatibility for software that requires usage of Mbed TLS crypto toolbox functions prefixed with ``mbedtls_``.

> .. caution::
>    Mbed TLS Legacy crypto toolbox APIs are marked as deprecated in nRF Connect SDK version 2.8.0, and will be removed in a future version. It is not recommended to make use of crypto toolbox functions prefixed with ``mbedtls_`` for any new designs. Please use equivalent functionality from PSA crypto APIs, instead.
>    Setting the :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` Kconfig option will enable the :kconfig:option:`MBEDTLS_LEGACY_CRYPTO_C` Kconfig option which will show a deprecation warning in the build output. Please use ::kconfig:option:`CONFIG_NRF_SECURITY` instead.

Reviewer: ```suggestion
.. caution::
   Mbed TLS legacy crypto toolbox APIs are marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
   It is not recommended to use crypto toolbox functions prefixed with ``mbedtls_`` for any new designs.
   Use the equivalent functionality from PSA crypto APIs instead.
   
   Setting the Kconfig option :kconfig:option:`CONFIG_NORDIC_SECURITY_BACKEND` will also enable the Kconfig option :kconfig:option:`MBEDTLS_LEGACY_CRYPTO_C`, which will show a deprecation warning in the build output.
   Use the Kconfig option :kconfig:option:`CONFIG_NRF_SECURITY` instead.
```
